src/bun.js/bindings/KeyObject.cpp

```diff
@@ -130,53 +130,119 @@ struct AsymmetricKeyValueWithDER {
 
 class KeyPassphrase {
 private:
-    WTF::CString pass_holder;
+    union {
+        WTF::CStringBuffer* strBuffer;
+        JSUint8Array* array;
+    } u;
+    uint8_t tag;
 
 public:
-    char* passphrase;
-    size_t passphrase_len;
-    bool hasPassphrase;
+    bool hasPassphrase()
+    {
+        return tag > 0;
+    }
+
+    char* data()
+    {
+        switch (tag) {
+        case 1: {
+            if (this->u.strBuffer) {
+                return const_cast<char*>(this->u.strBuffer->data());
```

Comment:
Is this `const_cast` safe? If we were to write to it and the underlying data was const we'd introduce UB

Reply:
Is safe internally is actually mutable data, and the data is not actually mutate in any manner anyway